tests/phpunit/Unit/support/renderFormFeaturedImageAndCaption.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ *  Tests for render_form_featured_image_and_caption()
+ *
+ * @since      1.0.0
+ * @author     Robert A. Gadon
+ * @package    spiralWebDb\ExtendGiveWP\Tests\Unit
+ * @link       http://spiralwebdb.com
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\ExtendGiveWP\Tests\Unit;
+
+use Brain\Monkey\Functions;
+use spiralWebDb\ExtendGiveWP\tests\phpunit\Unit\TestCase;
+use function spiralWebDb\ExtendGiveWP\render_form_featured_image_and_caption;
+
+/**
+ * Class Tests_RenderFormFeaturedImageAndCaption
+ *
+ * @covers ::\spiralWebDb\ExtendGiveWP\render_form_featured_image_and_caption
+ *
+ * @group   extend-give-wp
+ * @group   support
+ *
+ * phpcs:disable Squiz.Commenting.FunctionComment.MissingParamTag
+ */
+class Tests_RenderFormFeaturedImageAndCaption extends TestCase {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		require_once EXTEND_GIVE_WP_ROOT_DIR . '/src/support/load-assets.php';
+	}
+
+	/**
+	 * Test should render form featured image and caption given $form_id and image size.
+	 *
+	 * @dataProvider addTestData
+	 */
+	public function test_should_render_form_featured_image_and_caption( $post_data, $html, $expected_view ) {
+		Functions\expect( 'get_give_donation_form_id' )
+			->once()
+			->with( 'form_id' )
+			->andReturn( $post_data['form_id'] );
+		Functions\expect( 'get_option' )
+			->once()
+			->with( 'extend-give-wp', [] )
+			->andReturn( $post_data['attachment_id'] );
+		Functions\expect( 'get_post_field' )
+			->once()
+			->with( 'post_excerpt', 'attachment_id' )
+			->andReturn( $post_data['post_excerpt'] );
+		Functions\expect( 'wp_get_attachment_image' )
+			->once()
+			->with( $post_data['attachment_id'], 'large', false, [ 'class' => 'featured-image' ] )
+			->andReturn( $html );
+		Functions\expect( '_get_plugin_dir' )->andReturn( EXTEND_GIVE_WP_ROOT_DIR );
+		Functions\when( 'esc_attr')->justEcho();
```

Comment:
`esc_attr` does not echo, it returns the first argument after cleaning it. You can stub it and others by adding this code to your test case https://github.com/wp-media/phpunit/blob/master/Unit/TestCase.php#L63

Reply:
@hellofromtonya Per your comment on PR#16, I added `$this->setup_common_wp_stubs()` to the `setUp()` method in the unit test class. That resolved the problem with `esc_attr()` being undefined.